pkg/kubelet/cm/memorymanager/memory_manager.go

```diff
@@ -0,0 +1,151 @@
+
+	// GetPodTopologyHints implements the topologymanager.HintProvider Interface
+	// and is consulted to achieve NUMA aware resource alignment among this
+	// and other resource controllers.
+	GetPodTopologyHints(*v1.Pod) map[string][]topologymanager.TopologyHint
+}
+
+type manager struct {
+	sync.Mutex
+	policy Policy
+
+	// state allows to restore information regarding memory allocation for guaranteed pods
+	// in the case of the kubelet restart
+	state state.State
+
+	// containerRuntime is the container runtime service interface needed
+	// to make UpdateContainerResources() calls against the containers.
+	containerRuntime runtimeService
+
+	// activePods is a method for listing active pods on the node
+	// so all the containers can be updated in the reconciliation loop.
+	activePods ActivePodsFunc
+
+	// podStatusProvider provides a method for obtaining pod statuses
+	// and the containerID of their containers
+	podStatusProvider status.PodStatusProvider
+
+	// containerMap provides a mapping from (pod, container) -> containerID
+	// for all containers a pod
+	containerMap containermap.ContainerMap
+
+	nodeAllocatableReservation v1.ResourceList
+
+	// sourcesReady provides the readiness of kubelet configuration sources such as apiserver update readiness.
+	// We use it to determine when we can purge inactive pods from checkpointed state.
+	sourcesReady config.SourcesReady
+
+	// stateFileDirectory holds the directory where the state file for checkpoints is held.
+	stateFileDirectory string
+}
+
+var _ Manager = &manager{}
+
+// NewManager returns new instance of the memory manager
+func NewManager(reconcilePeriod time.Duration, machineInfo *cadvisorapi.MachineInfo, nodeAllocatableReservation v1.ResourceList, stateFileDirectory string, affinity topologymanager.Store) (Manager, error) {
+
+}
+
+// Start starts the memory manager reconcile loop under the kubelet to keep state updated
+func (m *manager) Start(activePods ActivePodsFunc, sourcesReady config.SourcesReady, podStatusProvider status.PodStatusProvider, containerRuntime runtimeService, initialContainers containermap.ContainerMap) error {
+
+}
+
+// AddContainer saves the value of requested memory for the guranteed pod under the state and set memory affinity according to the topolgy manager
+func (m *manager) AddContainer(p *v1.Pod, c *v1.Container, containerID string) error {
+
+}
+
+// Allocate is called to pre-allocate memory resources during Pod admission.
+func (m *manager) Allocate(pod *v1.Pod, container *v1.Container) error {
+
+}
+
+// RemoveContainer removes the container from the state
+func (m *manager) RemoveContainer(containerID string) error {
+
+}
+
+// State returns the state of the manager
+func (m *manager) State() state.Reader {
+	return m.state
+}
+
+// GetTopologyHints returns the topology hints for the topology manager
+func (m *manager) GetTopologyHints(pod *v1.Pod, container *v1.Container) map[string][]topologymanager.TopologyHint {
+
+}
```

Comment:
It implemented under the separate commit, I wanted to save commit history because it was implemented by @pablitoergosum and I wanted to save this contribution.